js/src/common/helpers/listItems.tsx

```diff
@@ -42,38 +49,43 @@ function withoutUnnecessarySeparators<Attrs>(items: ModdedVnode<Attrs>[]): Modde
  * By default, this tag is an `<li>` tag, but this is customisable through the
  * second function parameter, `customTag`.
  */
-export default function listItems<Attrs extends Record<string, unknown>>(
-  rawItems: ModdedVnode<Attrs> | ModdedVnode<Attrs>[],
-  customTag: string | Component<Attrs> = 'li',
+export default function listItems<Attrs extends ComponentAttrs>(
+  rawItems: ModdedChildrenWithItemname[],
+  customTag: string | (new () => Component<Attrs>) = 'li',
```

Comment:
I'd prefer to use [this](https://github.com/flarum/core/pull/3140#discussion_r753850972) instead of the union type.